### PR TITLE
[framework] Allow to run tests with test database only

### LINKED ISSUE #1039: I want to be able to run all tests/standards without the use of no-test database
### What is happening

I have a problem to run test phing targets with empty db-create database, i needed to modified parameters.yml database_name to shopsys-test to be able to generate error pages and other things like grunt and so on
### Expected result

I want to launch tests without the need to launch db-demo phing target but just test-db-demo phing target. So build test checks will be faster processed.

### PR DESCRIPTION
**Pull request by henzigo**
| Q             | A
| ------------- | ---
|Description, reason for the PR| The tests are very time consuming because we need to build whole application on CI server. With this upgrade, you will be able to run tests only with test database. That will decrease time-consuming for test stage.
|New feature| No 
|[BC breaks](https://docs.shopsys.com/en/latest/contributing/backward-compatibility-promise/)| Yes/No 
|Fixes issues| closes #1039 
|Have you read and signed our [License Agreement for contributions](https://www.shopsys.com/license-agreement)?| Yes

### CODE REVIEW

File: packages/framework/build.xml [hunk 1]
```diff
@@ -1056,7 +1064,7 @@
         </phingcall>
     </target>
 
-    <target name="tests" depends="npm-install-dependencies,error-pages-generate,test-db-demo,test-elasticsearch-index-recreate,test-elasticsearch-export,tests-unit,tests-functional,tests-smoke" description="Runs unit, functional and smoke tests. Builds new test database in the process."/>
+    <target name="tests" depends="npm-install-dependencies,test-db-demo,test-error-pages-generate,test-elasticsearch-index-recreate,test-elasticsearch-export,tests-unit,tests-functional,tests-smoke" description="Runs unit, functional and smoke tests. Builds new test database in the process."/>
```
Comment (grossmannmartin) [inline packages/framework/build.xml:1067]: When you now run the whole build (`build-demo-dev`), you will end with generated error pages from test DB, or?

**Comment by TomasLudvik** [comment]
Hello @henzigo, do you intend to finish this draft or you no longer need it and it can be closed?

**Comment by henzigo** [comment]
Hello @TomasLudvik, I'm glad that you are asking. I will try to finish this PR by the end of this week. Anyway, this is not what I need to be done, but as a Shopsys Framework developer, you should be interested to do this thing and not only close this.

**Comment by henzigo** [comment]
@TomasLudvik Tests are failing because of some bug in master.

File: packages/framework/build.xml (hunk 1 above)
Comment (henzigo) [reply packages/framework/build.xml]: Each environment has its own folder for storing generated HTML files

**Comment by TomasLudvik** [comment]
Hello @henzigo,

there is so much work to be done that not finished drafts are not in my top priorities and it depends on its authors if they want to finish it or not. In this case I thank you for finishing your work so we can decide in near future in what patch we will include this. 

There is no bug in master - PHPStan released new version that adds new control for its definition in PHPStan, there is already PR #1942 to solve it.

File: .travis.yml [hunk 2]
```diff
@@ -15,5 +15,10 @@ install:
...
 
-script:
-    - docker-compose exec php-fpm ./phing db-create test-db-create build-demo-dev tests-acceptance
+jobs:
+    include:
+        -   stage: Tests
+            name: Build all
+            script: docker-compose exec php-fpm ./phing db-create test-db-create build-demo-dev tests-acceptance
+        -   name: Build tests only
+            script: docker-compose exec php-fpm ./phing test-db-create tests tests-acceptance
```
Comment (grossmannmartin) [inline .travis.yml:24]: Why do you run tests twice? (part of build-demo-dev build and separately)

File: packages/framework/build.xml [hunk 3]
```diff
@@ -1010,6 +1010,14 @@
 
     <target name="test-elasticsearch-index-recreate" depends="production-protection,test-elasticsearch-index-delete,test-elasticsearch-index-create" description="Recreates structure for searching via elasticsearch in test environment (deletes existing structure and creates new one)"/>
 
+    <target name="test-error-pages-generate" depends="redis-check" description="Generates error pages displayed in production environment.">
```
Comment (grossmannmartin) [inline packages/framework/build.xml:1013]:
```suggestion
    <target name="test-error-pages-generate" depends="redis-check" description="Generates error pages for testing environment.">
```

File: project-base/config/paths.yaml [hunk 4]
```diff
@@ -1,6 +1,5 @@
 parameters:
     shopsys.content_dir_name: 'content'
-    shopsys.css_version_filepath: '%shopsys.web_dir%/assets/cssVersion'
```
Comment (grossmannmartin) [inline project-base/config/paths.yaml:3]: this is not related to a scope of this PR and I suggest to remove the change to avoid confusion

File: .travis.yml (hunk 2 above)
Comment (henzigo) [reply .travis.yml:24]: Because I want to test if tests are working without "production" database. But we could build application in first stage and tests in second stage. But this is on you guys.

File: project-base/config/paths.yaml (hunk 4 above)
Comment (henzigo) [reply project-base/config/paths.yaml:3]: That's fair point

File: .travis.yml (hunk 2 above)
Comment (grossmannmartin) [reply .travis.yml:24]: I understand and I agree with you – this new feature should be tested to ensure this will work in the future.

Do you need to somehow prepare the application before `test-db-create tests tests-acceptance` is run? 
If not, What about splitting the jobs and run standards in the first stage and tests in the second, as you did on the internal GitLab server?
That way it could run parallel and faster, or am I mistaken?

**Comment by henzigo** [comment]
It's already implemented.
